src/python/pants/option/global_options.py

```diff
@@ -784,8 +784,12 @@ def register_bootstrap_options(cls, register):
             type=int,
             default=2 ** 30,
             help=(
-                "The maximum memory usage of a pantsd process (in bytes). There is at most one "
-                "pantsd process per workspace."
+                "The maximum memory usage of the pantsd process (in bytes).\n\n"
+                "When the maximum memory is exceeded, the daemon will restart gracefully, "
+                "although all previous in-memory caching will be lost. Setting too low means that "
+                "you may miss out on some caching, whereas setting too high may over-consume "
+                "resources and may result in Pantsd being OOM-killed (out of memory).\n\n"
```

Comment:
OOM-kill is a bit jargon-y. Maybe "killed by the operating system due to that memory overconsumption" ?

Reply:
Fair. It is google-able, so I definitely want to keep the phrase, but can do so in parentheses.

Reply:
👍